custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js

```diff
@@ -285,8 +287,65 @@ describe('Download Directive', function () {
         }));
 
         it('tests that all users have access to ISSNIP monthly register', function () {
+            var expected = 9;
+            if (controller.haveAccessToFeatures) {
+                expected++;
+            }
             var length = controller.indicators.length;
-            assert.equal(9, length);
+            assert.equal(expected, length);
+        });
+
+        it('tests first possible data choice on THR raport', function () {
+            var fakeDate = new Date(2020, 8, 1);
+            var clock = sinon.useFakeTimers(fakeDate);
+            var expectedMonth = 7;
+            var expectedFirstYear = 2019;
+
+            controller.selectedYear = 2019;
+            controller.selectedIndicator = 10;
+            controller.onIndicatorSelect();
+
+            var firstAvailableMonth = controller.months[0].id;
+            var actualFirstYear = controller.selectedYear;
+
+            assert.equal(expectedMonth, firstAvailableMonth);
+            assert.equal(expectedFirstYear, actualFirstYear);
+            clock.restore();
+        });
+
+        it('tests latest possible data choice on THR raport', function () {
+            var fakeDate = new Date(2023, 8, 15);
```

Comment:
I am quite confused by this test, what is it doing?

Reply:
@calellowitz 
It checks if current month and year are available in dropbox. For example when current month is August  it assures that August is also possible selection.